content/en/docs/develop/porting.md

```diff
@@ -614,23 +614,328 @@ To make a patch:
+
+### Makefile
+
+The Makefile is generally short and simple and might remind you to Linux kernel modules that are built off-tree.
+For most applications the Makefile should contain no more than the following:
+
+```Makefile
+UK_ROOT  ?= $(PWD)/../../unikraft
+UK_LIBS  ?= $(PWD)/../../libs
+UK_PLATS ?= $(PWD)/../../plats
+LIBS  := $(UK_LIBS)/lib1:$(UK_LIBS)/lib2:$(UK_LIBS)/libN
+PLATS ?=
+
+all:
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS)
+
+$(MAKECMDGOALS):
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS) $(MAKECMDGOALS)
+```
+
+You can notice that the `Makefile` is just a wrapper around the [`Unikraft core Makefile`](https://github.com/unikraft/unikraft/blob/staging/Makefile).
+You can get the same restult if you use:
+
+```console
+$ make -C ../../unikraft A=. L=../../libs/lib1:../../libs/lib2:../../libs/libN
+```
+
+### Makefile.uk
+
+The Unikraft build system provides a number of functions and macros to make it easier to write the `Makefile.uk` file.
+Also, please ensure that all variables that you define begin with `APP[NAME]_` or `LIB[NAME]_` (e.g., `APPHELLOWORLD_`) so that they are properly namespaced.
+
+The first thing to do is to call the Unikraft `addlib` function to register the application with the system (note the letters "lib": everything in Unikraft is ultimately a library).
+This function call will also populate `APPNAME_BASE` (containing the directory path to the application sources) and `APPNAME_BUILD` (containing the directory path to the application's build directory):
+
+```Makefile
+$(eval $(call addlib,appname))
+```
+
+where `name` would be replaced by your application's name.
+In case your main application code should be downloaded as archive from a remote server, the next step is to set up a variable to point to a URL with the application sources (or objects, or pre-linked libraries, see further below), and to call Unikraft's `fetch` command to download and uncompress those sources (`APPNAME_ORIGIN` is populated containing the directory path to the extracted files)
+
+```Makefile
+APPNAME_ZIPNAME = myapp-v0.0.1
+APPNAME_URL = http://app.org/$(APPNAME_ZIPNAME).zip
+$(eval $(call fetch,appname,$(APPNAME_URL)))
+```
+
+Next we set up a call to use Unikraft's patch functionality.
+Even if you don't have any patches yet, it's good to have this set up in case you need it later.
+
+```Makefile
+APPNAME_PATCHDIR=$(APPNAME_BASE)/patches
+$(eval $(call patch,appname,$(APPNAME_PATCHDIR),$(APPNAME_ZIPNAME)))
+```
+
+With all of this in place, you can already start testing things out:
+
+```console
+$ make menuconfig
+[choose appropriate options and save configuration, see user's guide]
+$ make
+```
+
+You should see Unikraft downloading the application archive, unpacking it and doing the same plus building for any libraries you might have specified in the `Makefile` or through the menu (there will be nothing to build for your application yet as we haven't yet specified any sources to build).
```

Comment:
```suggestion
You should see Unikraft downloading the application archive and unpacking it.
It will do the same for libraries specified in the `Makefile` or through the menu.
Libraries will then be built.
There will be nothing to build for your application yet as we haven't yet specified any sources to build.
```